core/src/main/java/org/apache/iceberg/BaseScan.java

```diff
@@ -256,4 +265,95 @@ private static Schema lazyColumnProjection(TableScanContext context, Schema sche
   public ThisT metricsReporter(MetricsReporter reporter) {
     return newRefinedScan(table(), schema(), context().reportWith(reporter));
   }
+
+  private Optional<Long> adaptiveSplitSize(long tableSplitSize) {
+    if (!PropertyUtil.propertyAsBoolean(
```

Comment:
If this can access table properties, why read `tableSplitSize` outside and pass it in?

Reply:
We can move that in, it was just already being defined outside of this method.